pkg/client/v1/nimbleos/access_control_record.go

```diff
@@ -4,33 +4,51 @@ package nimbleos
 
 // AccessControlRecord - Manage access control records for volumes.
 // Export AccessControlRecordFields for advance operations like search filter etc.
-var AccessControlRecordFields *AccessControlRecord
+var AccessControlRecordFields *AccessControlRecordStringFields
 
 func init() {
 	IDfield := "id"
+	ApplyTofield := "apply_to"
 	ChapUserIdfield := "chap_user_id"
 	ChapUserNamefield := "chap_user_name"
 	InitiatorGroupIdfield := "initiator_group_id"
 	InitiatorGroupNamefield := "initiator_group_name"
+	Lunfield := "lun"
 	VolIdfield := "vol_id"
 	VolNamefield := "vol_name"
+	VolAgentTypefield := "vol_agent_type"
 	PeIdfield := "pe_id"
 	PeNamefield := "pe_name"
+	PeLunfield := "pe_lun"
 	SnapIdfield := "snap_id"
 	SnapNamefield := "snap_name"
+	PeIdsfield := "pe_ids"
+	Snaplunsfield := "snapluns"
+	CreationTimefield := "creation_time"
+	LastModifiedfield := "last_modified"
+	AccessProtocolfield := "access_protocol"
 
-	AccessControlRecordFields = &AccessControlRecord{
+	AccessControlRecordFields = &AccessControlRecordStringFields{
 		ID:                 &IDfield,
+		ApplyTo:            &ApplyTofield,
```

Comment:
Now that we've broken the relationship with the original AccessControlRecord struct, there doesn't seem to be a reason to use pointers anymore.  If we change all the fields from *string to string, we wouldn't need the temporary variables.  We can just use string literals inline here.

Reply:
Note to other reviewers: Making the values pure strings without pointers will break drop-in compatibility.  Upgrading to the new SDK version will require removing '*' characters where the fields are used (such as in examples/volume.go above).  I believe this is an appropriate trade-off because:

1. The field list used in the request parameters is of type []string.  These fields are defined mainly for that use.  We shouldn't require pointer dereferences to use them.
2. Existing use of the fields specifier was largely degraded prior to this fix, since it could only be used for string fields.  Therefore the impact might be minimal.

Reply:
Ah that's nice!

Reply:
I think the change is worth it. We can add that this a breaking feature for existing SDK users and how to overcome it. Will let others chime in as well